### ISSUE TITLE
Add support for running commands before playing the project from the editor

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#12191]
When using GDNative, it is often desired to compile the attached library (or libraries) before starting up the project (e.g. by pressing F5 in the editor).

Adding support for defining one or several commands to run before playing the project will make using GDNative more convenient (and there would be less risk to forget recompiling the library, thus wondering why things don't work as expected).

However, should the commands be stored in the editor, or in the project, considering that some of the user-defined commands may not work in a cross-platform manner (which is a bad thing for collaborative projects)?

**Comment by TeddyDD** [comment]
Godot has [feature tags](http://docs.godotengine.org/en/latest/learning/workflow/export/feature_tags.html) that could be used to store appropriate command per platform

**Comment by neikeq** [comment]
Maybe what you want is something like EditorNode's build callback to be exposed to scripts?

**Comment by juan-garcia-m** [comment]
This could be done by assigning an script instead.
This way we can call the external script any way we want, for example by calling an executable or by connecting to a server, etc.
Instead of having to press F5 it could also be done automatically on resource change, this way the file resource can be provided to the script.

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot-proposals/issues/119, as feature proposals are now tracked in the Godot proposals repository.

**Comment by willnationsdev** [comment]
FYI: This is actually what EditorPlugin::build() is for, iirc. Every EditorPlugin's `build()` function is called when the editor tries to run a scene. All of them have to return `true` or else the editor stops running the scene prematurely. This allows you to create an EditorPlugin to run GDNative build commands and the like. I expect to make this a more integrated component of my godotengine/godot-proposals#119 implementation.
